compiler/rustc_passes/src/dead.rs

```diff
@@ -1188,19 +1187,14 @@ fn check_mod_deathness(tcx: TyCtxt<'_>, module: LocalModDefId) {
         let def_kind = tcx.def_kind(item.owner_id);
 
         let mut dead_codes = Vec::new();
-        // if we have diagnosed the trait, do not diagnose unused methods
-        if matches!(def_kind, DefKind::Impl { .. })
+        // only diagnose unused assoc items for inherient impl and used trait
```

Comment:
```suggestion
        // Only diagnose unused assoc items for inherent impls,
```

The "and used trait for unused assoc items in impls of trait" is ungrammatical (?) and unclear.